scripts/liquidation_bot/liquidator-abi.ts

```diff
@@ -0,0 +1,232 @@
+    'type': 'function'
+  },
+  {
+    'inputs': [
+      {
+        'components': [
+          {
+            'internalType': 'address[]',
+            'name': 'accounts',
+            'type': 'address[]'
+          },
+          {
+            'internalType': 'address',
+            'name': 'pairToken',
+            'type': 'address'
+          },
+          {
+            'internalType': 'uint24',
+            'name': 'poolFee',
+            'type': 'uint24'
+          },
+          {
+            'internalType': 'bool',
+            'name': 'reversedPair',
+            'type': 'bool'
+          }
+        ],
+        'internalType': 'struct Liquidator.FlashParams',
+        'name': 'params',
+        'type': 'tuple'
+      }
+    ],
+    'name': 'initFlash',
+    'outputs': [],
+    'stateMutability': 'nonpayable',
+    'type': 'function'
+  },
+  {
+    'inputs': [
+      {
+        'internalType': 'address',
+        'name': '',
+        'type': 'address'
+      }
+    ],
+    'name': 'poolFees',
+    'outputs': [
+      {
+        'internalType': 'uint24',
+        'name': '',
+        'type': 'uint24'
+      }
+    ],
+    'stateMutability': 'view',
+    'type': 'function'
+  },
+  {
+    'inputs': [],
+    'name': 'refundETH',
+    'outputs': [],
+    'stateMutability': 'payable',
+    'type': 'function'
+  },
+  {
+    'inputs': [],
+    'name': 'swapRouter',
+    'outputs': [
+      {
+        'internalType': 'contract ISwapRouter',
+        'name': '',
+        'type': 'address'
+      }
+    ],
+    'stateMutability': 'view',
+    'type': 'function'
+  },
+  {
+    'inputs': [
+      {
+        'internalType': 'address',
+        'name': 'token',
+        'type': 'address'
+      },
+      {
+        'internalType': 'uint256',
+        'name': 'amountMinimum',
+        'type': 'uint256'
+      },
+      {
+        'internalType': 'address',
+        'name': 'recipient',
+        'type': 'address'
+      }
+    ],
+    'name': 'sweepToken',
+    'outputs': [],
+    'stateMutability': 'payable',
+    'type': 'function'
+  },
+  {
+    'inputs': [
+      {
+        'internalType': 'uint256',
+        'name': 'fee0',
+        'type': 'uint256'
+      },
+      {
+        'internalType': 'uint256',
+        'name': 'fee1',
+        'type': 'uint256'
+      },
+      {
+        'internalType': 'bytes',
+        'name': 'data',
+        'type': 'bytes'
+      }
+    ],
+    'name': 'uniswapV3FlashCallback',
+    'outputs': [],
+    'stateMutability': 'nonpayable',
+    'type': 'function'
+  },
+  {
+    'inputs': [
+      {
+        'internalType': 'uint256',
+        'name': 'amountMinimum',
+        'type': 'uint256'
+      },
+      {
+        'internalType': 'address',
+        'name': 'recipient',
+        'type': 'address'
+      }
+    ],
+    'name': 'unwrapWETH9',
+    'outputs': [],
+    'stateMutability': 'payable',
+    'type': 'function'
+  },
+  {
+    'stateMutability': 'payable',
+    'type': 'receive'
+  }
+];
```

Comment:
Curious why we needed to include all these abis here like this and they can't be generated with the normal hardhat compilation?

Reply:
yep, good call. we can use `getContractAt`.

#481

Reply:
what about all the other abis? I guess those aren't easy to access from here

Reply:
`getContractAt` uses the name ("Liquidator") to grab the ABI from the hardhat artifact.

The other ABIs are for tokens (COMP, LINK, UNI, etc). We don't have the contract code for those tokens locally, so I don't think we can use `getContractAt` to grab their ABIs.

I did some experimenting with using the local ERC20 interface for the ABI. So I replaced this:

```
const mockDai = new ethers.Contract(DAI, daiAbi, signer);
```

...with this:

```
const mockDai = await ethers.getContractAt("ERC20", DAI, signer) as ERC20;
```

This has some unintended consequences, though. It changes the return types of some of the methods. And I'm assuming it means you would be unable to call any non-ERC20 method on that contract.

It seems inaccurate, too. It assumes that all of the tokens have implemented the ERC20 interface in the exact same way.